### PR TITLE
[pg] Re-export 'DatabaseError' from 'pg-protocol'

### PR DESCRIPTION
**Pull request by cakoose**
Added in 'pg' v8.6.0
- ChangeLog entry: https://github.com/brianc/node-postgres/blob/master/CHANGELOG.md#pg860
- Commit: https://github.com/brianc/node-postgres/commit/4b229275cfe41ca17b7d69bd39f91ada0068a5d0

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/pg/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for pg 7.14
+// Type definitions for pg 8.6
```
Comment (HoldYourWaffle) [inline types/pg/index.d.ts:1]: Is this a full update to pg v8?

File: types/pg/package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
     "private": true,
     "dependencies": {
         "pg-types": "^2.2.0",
-        "pg-protocol": "^1.2.0"
+        "pg-protocol": "*",
+        "pg-types": "^2.2.0"
```
Comment (HoldYourWaffle) [inline types/pg/package.json:6]: `pg-types` is listed twice now.

**Review by HoldYourWaffle — CHANGES_REQUESTED** [review]

File: types/pg/package.json (hunk 2 above)
Comment (cakoose) [reply types/pg/package.json:6]: Oh whoops, good catch.

File: types/pg/index.d.ts (hunk 1 above)
Comment (cakoose) [reply types/pg/index.d.ts:1]: The new `DatabaseError` symbol was added in 8.6.0.  I'm actually not sure what to do with the version number in this file.

Comment (HoldYourWaffle) [reply types/pg/index.d.ts:1]: Hmm I'm not entirely sure either.
Could you perhaps investigate what other updates would be necessary to get these definitions up to speed with v8?

Comment (cakoose) [reply types/pg/index.d.ts:1]: A quick glance at the [changelog](https://github.com/brianc/node-postgres/blob/master/CHANGELOG.md) doesn't show any breaking changes to the types.  I don't have the time to do a more thorough check at the moment.

**Review by HoldYourWaffle — APPROVED** [review]
Ah right, I forgot about this. Sorry!

Based on a quick glance at the changelog there are quite a few items missing to bring these definitions fully up-to-date with v8.6. The only breaking change I could spot was making `Pool` and `Query` ES6 classes, but that already seems to be the case in the definitions.

I suppose updating everything is not on you though, so I think it's fine to merge this now.

**Comment by cakoose** [comment]
@HoldYourWaffle: Thanks.  As long as there aren't any incompatibilities with 8.6 I think it's somewhat normal for DefinitelyTyped definitions to be incomplete, with contributors adding new definitions as they need them.

**Comment by cakoose** [comment]
Ready to merge

**Pull request merged**